folding-schemes/src/folding/hypernova/circuits.rs

```diff
@@ -345,24 +430,320 @@ fn compute_c_gadget<F: PrimeField>(
     Ok(c)
 }
 
+#[derive(Debug, Clone)]
+pub struct AugmentedFCircuit<
+    C1: CurveGroup,
+    C2: CurveGroup,
+    GC2: CurveVar<C2, CF2<C2>>,
+    FC: FCircuit<CF1<C1>>,
+> where
+    for<'a> &'a GC2: GroupOpsBounds<'a, C2, GC2>,
+{
+    pub _c2: PhantomData<C2>,
+    pub _gc2: PhantomData<GC2>,
+    pub poseidon_config: PoseidonConfig<CF1<C1>>,
+    pub ccs: CCS<C1::ScalarField>, // CCS of the AugmentedFCircuit
+    pub i: Option<CF1<C1>>,
+    pub i_usize: Option<usize>,
+    pub z_0: Option<Vec<C1::ScalarField>>,
+    pub z_i: Option<Vec<C1::ScalarField>>,
+    pub external_inputs: Option<Vec<C1::ScalarField>>,
+    pub u_i_C: Option<C1>, // u_i.C
+    pub U_i: Option<LCCCS<C1>>,
+    pub U_i1_C: Option<C1>, // U_{i+1}.C
+    pub F: FC,              // F circuit
+    pub x: Option<CF1<C1>>, // public input (u_{i+1}.x[0])
+    pub nimfs_proof: Option<NIMFSProof<C1>>,
```

Comment:
this is like the way of doing in arkworks, eg: [github.com/arkworks-rs/r1cs-tutorial/rollup/src/rollup.rs#L19](https://github.com/arkworks-rs/r1cs-tutorial/blob/5d3a9022fb6deade245505748fd661278e9c0ff9/rollup/src/rollup.rs#L19)
this is used so that when generating the constraints, we can do it without needing to define all the values for each of the fields, and internally the generate_constraints will use default values (eg. zeroes) for the fields. In this way, we can generate the constraints and obtain the R1CS matrices before starting to work with real values.